arrows/dbow2/CMakeLists.txt

```diff
@@ -39,10 +39,6 @@ set( dbow2_sources
   Timestamp.cxx
   )
 
-if( KWIVER_HAS_OPENCV_VER_3 )
-  add_definitions(-DKWIVER_HAS_OPENCV_VER_3)
```

Comment:
Do we need to replace this with the new `target_compile_definitions` approach used for `kwiver_algo_ocv`?  Or was this definition not actually needed?

Reply:
This popped up when I was looking for instances to replace, but dbow2 never references it (directly). It doesn't obviously reference it _indirectly_ either; cursory checking did not turn up any instances of the dbow2 arrow using the ocv arrow/algorithms. The ocv arrow _does_ use the OpenCV version in some headers, however, so I (just now) changed the `target_compile_definitions` for that to `PUBLIC`. Note that it's possible the *matlab* arrow has always been broken due to this; that one _does_ consume the ocv arrow (though I didn't dig deep enough to tell if it consumes any bits that behave differently between 2.x and 3.x/4.x).

Reply:
IOW, this was removed because AFAICT it was never needed. I suspect the dbow2 CMakeLists.txt originated as a copy of the ocv one.